### PR TITLE
fix: retry pre-stream rate limit SSE errors

### LINKED ISSUE #6643: fix: Responses via Chat does not retry code-only rate limit SSE errors
## 问题描述

`/v1/responses` 通过 Chat Completions 上游转发时，部分兼容上游会保持 HTTP 200，并在首个有效输出前通过 SSE data 返回限流错误。

当错误只包含 `code` 而没有 `type` 时，例如：

```text
data: {"error":{"message":"rate limited","code":"rate_limit_exceeded"}}
```

当前 `OaiChatToResponsesStreamHandler` 只有在 `OpenAIError.Type` 非空时才进入错误处理。因此该错误不会映射为 HTTP 429，也不会进入现有自动重试逻辑，随后还可能被当成普通 Chat chunk 解析。

## 复现

无需真实上游，在单元测试中给 handler 输入：

```text
data: {"error":{"message":"rate limited","code":"rate_limit_exceeded"}}
```

上游 HTTP 状态为 200，且下游尚未输出任何 Responses 事件。

## 预期结果

当错误的 `type` 或 `code` 任一明确表示 rate limit 时，包括：

* `rate_limit_error`
* `rate_limit_exceeded`
* `429`

应在下游尚未开始输出时映射为 HTTP 429，从而复用现有重试逻辑。

下游已开始输出时不应透明重试。

### PR DESCRIPTION
**Pull request by AmPlace**
# ⚠️ 提交说明 / PR Notice

> [!IMPORTANT]
>
> - 这份说明已按实际代码和测试结果整理；实现与测试过程使用了 Codex 辅助。

## 📝 变更描述 / Description

这个 PR 修复 `/v1/responses` 通过 Chat Completions 上游转发时，HTTP 200 SSE 在首个有效响应事件前返回限流错误却不会进入现有 429 重试逻辑的问题。

部分 Chat-compatible 上游会保持 HTTP 200，然后在 SSE data 中返回：

```json
{"error":{"type":"rate_limit_error","code":"rate_limit_exceeded","message":"rate limited"}}
```

此前转换 handler 直接沿用上游 HTTP 200 构造内部错误，`shouldRetry` 因状态码为 2xx 跳过重试。

本次只在以下条件同时满足时把内部状态恢复为 429：

- 上游 HTTP 状态为 2xx；
- 下游尚未输出任何 Responses SSE 事件；
- SSE 错误的 type/code 明确表示 rate limit。

这样可以复用现有重试配置和渠道选择逻辑。只要下游已经开始输出，就保持原状态且不做透明重试，避免重复或混合流内容。真实 HTTP 429 的处理逻辑没有改动。

## 🚀 变更类型 / Type of change

- [x] 🐛 Bug 修复 (Bug fix) - *请关联对应 Issue，避免将设计取舍、理解偏差或预期不一致直接归类为 bug*
- [ ] ✨ 新功能 (New feature) - *重大特性建议先通过 Issue 沟通*
- [ ] ⚡ 性能优化 / 重构 (Refactor)
- [ ] 📝 文档更新 (Documentation)

## 🔗 关联任务 / Related Issue

- Closes #6643

## ✅ 提交前检查项 / Checklist

- [x] **人工确认:** 我已检查并整理此描述，没有直接粘贴未经处理的输出。
- [x] **非重复提交:** 已搜索现有 Issues 和 PRs，没有发现完整处理该边界的提交。
- [ ] **Bug fix 说明:** 当前没有可关联的精确 Issue。
- [x] **变更理解:** 已确认只在下游未开始时恢复限流状态，已开始的流不会透明重试。
- [x] **范围聚焦:** 本 PR 未包含无关改动。
- [x] **本地验证:** 已运行相关测试。
- [x] **安全合规:** 代码中无敏感凭据，并符合现有代码规范。

## 📸 运行证明 / Proof of Work

```text
go test ./controller -run 'TestResponsesViaChat' -count=1 -v
PASS TestResponsesViaChatRetriesRateLimitErrorBeforeOutput
PASS TestResponsesViaChatRetriesCodeOnlyRateLimitBeforeOutput
PASS TestResponsesViaChatDoesNotRetryRateLimitAfterOutput

 go test ./relay/channel/openai ./controller -count=1
ok github.com/QuantumNous/new-api/relay/channel/openai
ok github.com/QuantumNous/new-api/controller

 git diff --check
(no output)
```

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of rate-limit errors during streaming responses.
  * Returns HTTP 429 when a rate-limit error occurs before output begins.
  * Preserves successful response behavior when output has already started.

* **Tests**
  * Added coverage for retry decisions and status codes across streaming rate-limit scenarios.

### CODE REVIEW

**Comment by AmPlace** [comment]
GitHub is currently rejecting both `markPullRequestReadyForReview` and replacement `CreatePullRequest` operations for the PR author with a permission error. The branch, tests, and checks are ready. Could a maintainer please mark this PR as ready for review?

**Comment by AmPlace** [comment]
由于当前仓库限制同一作者同时维护的开放 PR 数量，先暂时排队关闭。本分支、测试和提交会保留；该 PR 当前可合并，等现有高优先级修复释放名额后可以直接重新打开。

File: relay/channel/openai/responses_via_chat.go [hunk 1]
```diff
@@ -97,7 +98,8 @@ func OaiChatToResponsesStreamHandler(c *gin.Context, info *relaycommon.RelayInfo
 		var errorResp dto.OpenAITextResponse
 		if err := common.UnmarshalJsonStr(data, &errorResp); err == nil {
 			if oaiError := errorResp.GetOpenAIError(); oaiError != nil && oaiError.Type != "" {
-				streamErr = types.WithOpenAIError(*oaiError, resp.StatusCode)
+				statusCode := responsesViaChatStreamErrorStatus(resp.StatusCode, oaiError, c.Writer.Written())
+				streamErr = types.WithOpenAIError(*oaiError, statusCode)
```
[the remark this answers is not in the document]
Comment (AmPlace) [reply relay/channel/openai/responses_via_chat.go:102]: Addressed. The stream error path now processes any non-nil OpenAI error, allowing code-only rate-limit errors to reach the existing status mapper. A regression test covers an SSE error containing rate_limit_exceeded without a type field.

**Comment by AmPlace** [comment]
The code-only rate-limit edge case from review has been fixed and covered by a regression test. The PR is now linked to #6643.

Validation:
- go test ./controller -run 'TestResponsesViaChat' -count=1 -v
- go test ./relay/channel/openai ./controller -count=1
- git diff --check

The Actions run is currently action_required with no jobs. Could a maintainer please approve CI and review this PR?
